src/checkers/inference/solver/GeneralSolver.java

```diff
@@ -0,0 +1,392 @@
+            StatisticRecorder.record(StatisticKey.GRAPH_GENERATION_TIME, (graphBuildingEnd - graphBuildingStart));
+            solution = graphSolve(constraintGraph, configuration, slots, constraints, qualHierarchy,
+                    processingEnvironment, defaultSerializer);
+        } else {
+            realBackEnd = createBackEnd(backEndType, configuration, slots, constraints, qualHierarchy,
+                    processingEnvironment, lattice, defaultSerializer);
+            solution = solve();
+        }
+
+        if (solution == null) {
+            // Solution should never be null.
+            ErrorReporter.errorAbort("Null solution detected!");
+        }
+
+        if (collectStatistic) {
+            Map<String, Integer> modelRecord = recordSlotConstraintSize(slots, constraints);
+            PrintUtils.printStatistic(StatisticRecorder.getStatistic(), modelRecord, backEndType,
+                    useGraph, solveInParallel);
+            PrintUtils.writeStatistic(StatisticRecorder.getStatistic(), modelRecord, backEndType,
+                    useGraph, solveInParallel);
+        }
+        return solution;
+    }
+
+    /**
+     * This method configures following arguments: backEndType, useGraph,
+     * solveInParallel, and collectStatistic
+     * 
+     * @param configuration
+     */
+    private void configureSolverArgs(final Map<String, String> configuration) {
+
+        final String backEndName = configuration.get(SolverArg.backEndType.name());
+        final String useGraph = configuration.get(SolverArg.useGraph.name());
+        final String solveInParallel = configuration.get(SolverArg.solveInParallel.name());
+        final String collectStatistic = configuration.get(SolverArg.collectStatistic.name());
+
+        backEndType = backEndName == null ? BackEndType.MAXSAT : BackEndType.getBackEndType(backEndName);
+        if (backEndType == null) {
+            ErrorReporter.errorAbort("Back end \"" + backEndName + "\" has not been implemented yet.");
+        }
+
+        if (useGraph == null || useGraph.equals(Constants.TRUE)) {
+            // Configure use of constraint graph. Default is true.
+            this.useGraph = true;
+        } else {
+            this.useGraph = false;
+        }
+
+        if (backEndType.equals(BackEndType.LOGIQL)) {
+            // Configure solving strategy.
+            this.solveInParallel = false;
+        } else if (solveInParallel == null || solveInParallel.equals(Constants.TRUE)) {
+            this.solveInParallel = true;
+        } else {
+            this.solveInParallel = false;
+        }
+
+        if (collectStatistic == null || collectStatistic.equals(Constants.FALSE)) {
+            // Configure statistic collection.
+            this.collectStatistic = false;
+        } else if (collectStatistic.equals(Constants.TRUE)) {
+            this.collectStatistic = true;
+        }
+
+        // Sanitize the configuration if it needs.
+        sanitizeConfiguration();
```

Comment:
Maybe change the `confirgureSolverArgs()` visibility to protected, so that sub-classes can override this method, have customized configurations, also do sanitization there.